crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -102,7 +105,7 @@ pub struct DatabaseProvider<TX> {
     chain_spec: Arc<ChainSpec>,
     /// Snapshot provider
     #[allow(dead_code)]
-    snapshot_provider: Option<Arc<SnapshotProvider>>,
+    pub snapshot_provider: Option<Arc<SnapshotProvider>>,
```

Comment:
```suggestion
    pub(crate) snapshot_provider: Option<Arc<SnapshotProvider>>,
```

also no need for `dead_code`

Reply:
hmm, can we do `self.snapshot_provider.commit()` in `DatabaseProvider::commit`, so `DatabaseProviderRW::commit` just delegates to it as previously?